src/video-grid/VideoTile.tsx

```diff
@@ -242,44 +167,315 @@ export const VideoTile = forwardRef<HTMLDivElement, Props>(
+  displayName: string;
+  maximised: boolean;
+  fullscreen: boolean;
+  onToggleFullscreen: (itemId: string) => void;
+}
+
+const ScreenShareTile = subscribe<ScreenShareTileProps, HTMLDivElement>(
+  (
+    {
+      vm,
+      className,
+      style,
+      targetWidth,
+      targetHeight,
+      nameTag,
+      displayName,
+      maximised,
+      fullscreen,
+      onToggleFullscreen,
+    },
+    ref,
+  ) => {
+    const { t } = useTranslation();
+    const video = useStateObservable(vm.video);
+    const unencryptedWarning = useStateObservable(vm.unencryptedWarning);
+    const onClickFullScreen = useCallback(
+      () => onToggleFullscreen(vm.id),
+      [onToggleFullscreen, vm],
+    );
+
+    const FullScreenIcon = fullscreen ? CollapseIcon : ExpandIcon;
+
+    return (
+      <Tile
+        ref={ref}
+        className={classNames(className, styles.screenshare)}
+        style={style}
+        targetWidth={targetWidth}
+        targetHeight={targetHeight}
+        video={video}
+        member={vm.member}
+        videoEnabled={true}
+        maximised={maximised}
+        unencryptedWarning={unencryptedWarning}
+        nameTag={nameTag}
+        displayName={displayName}
+        primaryButton={
+          !vm.local && (
+            <button
+              aria-label={
+                fullscreen
+                  ? t("video_tile.full_screen")
+                  : t("video_tile.exit_full_screen")
+              }
+              onClick={onClickFullScreen}
+            >
+              <FullScreenIcon aria-hidden width={20} height={20} />
+            </button>
+          )
+        }
+      />
     );
   },
 );
 
+ScreenShareTile.displayName = "ScreenShareTile";
+
+interface Props {
+  vm: TileViewModel;
+  maximised: boolean;
+  fullscreen: boolean;
+  onToggleFullscreen: (itemId: string) => void;
+  onOpenProfile: () => void;
+  targetWidth: number;
+  targetHeight: number;
+  className?: string;
+  style?: ComponentProps<typeof animated.div>["style"];
+  showSpeakingIndicator: boolean;
+}
+
+export const VideoTile = forwardRef<HTMLDivElement, Props>(
+  (
+    {
+      vm,
+      maximised,
+      fullscreen,
+      onToggleFullscreen,
+      onOpenProfile,
+      className,
+      style,
+      targetWidth,
+      targetHeight,
+      showSpeakingIndicator,
+    },
+    ref,
+  ) => {
+    const { t } = useTranslation();
+
+    // Handle display name changes.
+    // TODO: Move this into the view model
+    const [displayName, setDisplayName] = useReactiveState(
+      () => vm.member?.rawDisplayName ?? "[👻]",
+      [vm.member],
+    );
+    useEffect(() => {
+      if (vm.member) {
+        const updateName = (): void => {
+          setDisplayName(vm.member!.rawDisplayName);
+        };
+
+        vm.member!.on(RoomMemberEvent.Name, updateName);
+        return (): void => {
+          vm.member!.removeListener(RoomMemberEvent.Name, updateName);
+        };
+      }
+    }, [vm.member, setDisplayName]);
```

Comment:
Why is this not part of the view model?
(Just saw its mentioned in the TODO. What is holding back moving it?)

Reply:
Hmm, I can't remember actually what the difficulty was! But regardless, I think it doesn't hurt to leave little things like this to be refactored later, in the spirit of working iteratively